node/caching-compiler.js

```diff
@@ -17,8 +17,17 @@ exports.compile = function (content, filename) {
     // Treat the REPL as if there was no filename.
     filename = null;
   }
-  var info = filename ? getPkgInfo(filename) : fallbackPkgInfo;
-  return info ? readWithCache(info, content) : content;
+  
+  try {
+    var info = filename ? getPkgInfo(filename) : fallbackPkgInfo;
+    return info ? readWithCache(info, content) : content;
+  } catch(e) {
+    if (filename) {
+      e.message += 'While processing file: ' + filename;
+    }
+    throw e;
+  }
+
 };
```

Comment:
Performance of this function is important, so perhaps we should split it into two different helper functions, with only one function containing a `try`-`catch` block (i.e., the function called when `filename` is provided).